src/webui/webapplication.cpp

```diff
@@ -736,6 +738,37 @@ void WebApplication::action_command_setLabel()
     }
 }
 
+void WebApplication::action_command_addLabel()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("label");
+
+    QString label = request().posts["label"].trimmed();
+    if (!Utils::Fs::isValidFileSystemName(label) && !label.isEmpty()) {
+        status(400, "Labels must not contain special characters");
+        return;
+    }
+
+    Preferences::instance()->addTorrentLabelExternal(label);
+}
+
+void WebApplication::action_command_removeLabels()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("labels");
+
+    QStringList labels = request().posts["labels"].split("|");
+
+    foreach (QString label, labels) {
```

Comment:
This is very minor, you can ignore it if you want, just a suggestion.

Since qbt support C++11 you can use range-based for loop, e.g. 
```c++
for (const QString &label : labels) {
```

or just change to use reference:
```c++
foreach (QString &label, labels) {
```